actions/java-maven-setup/action.yaml

```diff
@@ -0,0 +1,27 @@
+name: "Setup Maven"
+description: "Setup Java and Maven"
+
+inputs:
+  java-distribution:
+    description: "Java distribution to be installed. (Default is microsoft)"
+    required: false
+    default: "microsoft"
+  java-version:
+    description: "Java version to be installed. (Default is 11)"
+    required: false
+    default: "11"
+  maven-version:
+    description: "Maven version to be installed."
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-java@v3
+      with:
+        distribution: ${{ inputs.java-distribution }}
+        java-version: ${{ inputs.java-version }}
+
+    - name: Set up Maven
+      uses: yuzhiyongcn/setup-maven@v1.0.0
```

Comment:
Not a hard requirement for this PR, but does this action support caching in any way or do we want to have it in the future for Maven projects?

Reply:
Does not seem to be the case. I also don't want to overengineer this as we almost not use maven at all